framework/test/gtest/BundleManifestTest.cpp

```diff
@@ -187,7 +223,215 @@ TEST(BundleManifestTest, ParseManifest)
   ASSERT_EQ(any_cast<int>(m["number"]), 4);
   ASSERT_EQ(m["list"].Type(), typeid(std::vector<Any>));
   ASSERT_EQ(any_cast<std::vector<Any>>(m["list"]).size(), 2ul);
+}
+
+namespace cppmicroservices {
+
+struct TestBundleAService
+{
+  virtual ~TestBundleAService() {}
+};
+
+}
+
+TEST_F(BundleManifestTest, DirectManifestInstall)
+{
+#ifdef US_BUILD_SHARED_LIBS
+  auto ctx = framework.GetBundleContext();
+
+  cppmicroservices::AnyMap manifests(cppmicroservices::any_map::UNORDERED_MAP_CASEINSENSITIVE_KEYS);
+  cppmicroservices::AnyMap::unordered_any_cimap testBundleAManifest = {
+    { "bundle.symbolic_name", std::string("TestBundleA") }
+    , { "bundle.activator", true }
+  };
+  manifests["TestBundleA"] = cppmicroservices::AnyMap(testBundleAManifest);
+
+  auto const libPath = fullLibPath("TestBundleA");
+  auto const& bundles = ctx.InstallBundles(libPath, manifests);
+
+  ASSERT_EQ(1, bundles.size());
+  for (auto b : bundles) {
+    if (b.GetSymbolicName() != "TestBundleA") continue;
+    auto headers = b.GetHeaders();
+    auto manifest = cppmicroservices::any_cast<cppmicroservices::AnyMap>(manifests["TestBundleA"]);
+    
+    // check to make sure that all the headers in the manifest are there
+    for (auto m : manifest)
+    {
+      ASSERT_EQ(m.second.ToString(), headers[m.first].ToString());
+    }
+    break;
+  }
+#endif
+}
+
+TEST_F(BundleManifestTest, DirectManifestInstallBadLocation)
```

Comment:
there should be a test for passing in a manifest which doesn't have the required meta-data (i.e. bundle.symbolic_name)